storage/rocksdb/rdb_datadic.cc

```diff
@@ -3179,6 +3181,17 @@ uint32 Rdb_key_def::calculate_index_flag_offset(uint32 index_flags,
   return offset;
 }
 
+uint32 Rdb_key_def::calculate_total_index_flags_length(uint32 index_flags) {
```

Comment:
perhaps we should just reuse `calculate_index_flag_offset`, but pass in the `MAX_FLAG`.  This can get us the offset to where the actual record starts.  

```
  enum INDEX_FLAG {
    TTL_FLAG = 1 << 0,

    // MAX_FLAG marks where the actual record starts
    // This flag always needs to be set to the last index flag enum.
    MAX_FLAG = TTL_FLAG << 1,
  };
```